### DISCUSSION TITLE
Input a url as string , transcribe as normal after fetch on the url

### DISCUSSION THREAD

**Discussion by rowntreerob** [deepgram/community#657] [Feature Requests]
Usecase: 
   user visits, starts podcast playback, checks media-length and Wants a quick transcription.
   The page's DOM has the url. Simple click on a bookmarklet selects <audio.src=> from the DOM, using that url string in a call to ```client.prerecorded.v("1").transcribe_url()```
   
  [NYT bookmark let](https://gist.github.com/rowntreerob/a7d8c3d687cae644f5bfb00c6f347390)
  
  All this does is shorten the process. Explicit download of the audio.src is no longer req'd and the task to get the URL from any page is a single click , returning <audio src= > as a URL.   
  
![Screenshot 2024-03-21 12 02 27 PM](https://github.com/deepgram/community/assets/169984/a874fa99-c7f1-4d56-ad46-e95fa11e2599)

[video of an ](https://youtu.be/tFqp_C5NXJ4)  interaction using the feature.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@rowntreerob  I noticed you took a screen shot of our API Playground can you give us more details of what you are suggesting here?

**Comment by rowntreerob** [comment]
current :  you need to download the input file to localhost and then supply
the local file path to the fileUploader in the form

what i want is in addition to the file uploader , another option to supply
the URL to the AUDIO.

when the client requests transcription on a URL, the server just does a
FETCH instead of handling the form's uploaded file mentioned above .

**Comment by rowntreerob** [comment]
add delts:

https://gist.github.com/rowntreerob/a7d8c3d687cae644f5bfb00c6f347390
for a scriplet that i use all the time on news sites with alot of podcasts

code: finds the <audio> and getting the audio.src which is the url to the
media.

step 2 is it uses the url in just the way i suggested in my post for new
feature.
step 2 is just a proxy feeding the media url to a handler with
      code like here
<https://gist.github.com/rowntreerob/a0d29860aea8f66c0315648a2d8a1334#file-transcribemp3-py-L29>

**Comment by jpvajda** [comment]
@rowntreerob  Thanks for clarifying! I'll share this with the product team that works on API Playground for a future consideration.
